lib/kubernetes-deploy/runner.rb

```diff
@@ -106,6 +108,13 @@ def run
       phase_heading("Checking initial resource statuses")
       resources.each(&:sync)
 
+      ejson = EjsonSecretProvisioner.new(namespace: @namespace, context: @context, template_dir: @template_dir)
+      if ejson.secrets_requested?
```

Comment:
> I did it this way to be able to avoid printing the heading when there's nothing to do, i.e. hide the feature from the average deploy user.

I see that, still makes for fairly awkward code see how that check `secrets_requested?` repeats on the inner functions. and IMO breaks encapsulation, why not have the hypotetycal `run` return a value and print based on that?